app/Entity/ModelEntities/AnalysisEntities/AnalysisTool.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+namespace App\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\Common\Collections\Criteria;
+use Doctrine\ORM\Mapping as ORM;
+
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="analysis_tool")
+ * @ORM\DiscriminatorColumn(name="hierarchy_type", type="string")
+ */
+class AnalysisTool implements IdentifiedObject
+{
+    /**
+     * @ORM\Id
```

Comment:
id, name (also descriptions) appears in most of the entities, use abstract class or a trait